tests/functional/test_waiter_config.py

```diff
@@ -11,16 +11,94 @@
+                        "enum": ["api"]
+                    },
+                    "operation": {"type": "string"},
+                    "description": {"type": "string"},
+                    "delay": {
+                        "type": "number",
+                        "minimum": 0,
+                    },
+                    "maxAttempts": {
+                        "type": "integer",
+                        "minimum": 1
+                    },
+                    "acceptors": {
+                        "type": "array",
+                        "items": {
+                            "type": "object",
+                            "properties": {
+                                "state": {
+                                    "type": "string",
+                                    "enum": ["success", "retry", "failure"]
+                                },
+                                "matcher": {
+                                    "type": "string",
+                                    "enum": [
+                                        "path", "pathAll", "pathAny",
+                                        "status", "error"
+                                    ]
+                                },
+                                "argument": {"type": "string"},
+                                "expected": {
+                                    "oneOf": [
+                                        {"type": "string"},
+                                        {"type": "number"},
+                                        {"type": "boolean"}
+                                    ]
+                                }
+                            },
+                            "required": [
+                                "state", "matcher", "expected"
+                            ],
+                            "additionalProperties": False
+                        }
+                    }
+                },
+                "required": ["operation", "delay", "maxAttempts", "acceptors"],
+                "additionalProperties": False
+            }
+        }
+    },
+    "additionalProperties": False
+}
+
+
 def test_lint_waiter_configs():
     session = botocore.session.get_session()
+    validator = Draft4Validator(WAITER_SCHEMA)
     for service_name in session.get_available_services():
         client = session.create_client(service_name, 'us-east-1')
         service_model = client.meta.service_model
+        try:
+            # We use the loader directly here because we need the entire
+            # json document, not just the portions exposed (either
+            # internally or externally) by the WaiterModel class.
+            loader = session.get_component('data_loader')
+            waiter_model = loader.load_service_model(
+                service_name, 'waiters-2')
+        except UnknownServiceError:
+            # The service doesn't have waiters
+            continue
+        validator.validate(waiter_model)
```

Comment:
Forgot a comment... We may want to put this in a yield statement to make sure the entire test does not immediately fail for a single waiter. Also what does the error message look like for something a waiter config that fails this test?

Reply:
```json
{
  "version": 2,
  "waiters": {
    "BucketExists": {
      "delay": 5,
      "operation": "HeadBucket",
      "maxAttempts": 20,
      "acceptors": [
        {
          "expected": 200,
          "matcher": "status",
          "state": "success"
        },
        {
          "expected": 404,
          "matcher": "status",
          "state": "retry"
        }
      ]
    }
  }
}
```

```
Traceback (most recent call last):
  File "/Users/phjordon/.pyenv/versions/3.6.2/envs/bc/lib/python3.6/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/phjordon/workspace/botocore/tests/functional/test_waiter_config.py", line 137, in _validate_schema
    raise AssertionError('\n'.join(errors))
AssertionError: Additional properties are not allowed ('foo' was unexpected)
```